### PR TITLE
Create and Index PPM

### PR DESCRIPTION
**Pull request by macrael**
## Description

This adds the Create and Index PPM handlers and the PPM model. 

## Reviewer Notes

I tried a new pattern for ensuring that we only access moves in our handlers that the service member has access to, check out models/moves.go#GetMoveForUser

It appears to be a go anti-pattern to introspect error objects, so I'm returning a Result object that encapsulates the ways we expect this method to fail. It's possible it would be more clear/go-like to pull that out as a separate error object (kind of like how pop does verrs) so let me know what you think of that. 

You can see the difference in the usage of this pattern by looking at the changes I made to handlers/signed_certifications.go. With GetMoveForUser, it is more difficult for a handler to accidentally do something with a MoveID that the user doesn't have access to. I'd love feedback on this pattern. 

## Verification Steps

* [ ] All tests pass.
* [ ] The requirements listed in [Querying the Database Safely](https://github.com/transcom/mymove/blob/master/docs/backend.md#querying-the-database-safely) have been satisfied.
* [ ] There are no [aXe](https://www.deque.com/products/aXe/) warnings for UI.
* [ ] This works in IE.

## References

https://www.pivotaltracker.com/story/show/155909967

### CODE REVIEW

File: pkg/models/move.go [hunk 1]
```diff
@@ -56,11 +60,73 @@ func (m *Move) ValidateUpdate(tx *pop.Connection) (*validate.Errors, error) {
 	return validate.NewErrors(), nil
 }
 
-// GetMoveByID fetches a Move model by their database ID
-func GetMoveByID(db *pop.Connection, id uuid.UUID) (Move, error) {
-	move := Move{}
+// MoveResult is returned by GetMoveForUser and encapulates wether the call succeeded and why it failed.
```
Comment (amitch23) [inline pkg/models/move.go]: Spelling: "whether", "encapsulates"

File: pkg/models/move.go [hunk 2]
```diff
@@ -56,11 +60,73 @@ func (m *Move) ValidateUpdate(tx *pop.Connection) (*validate.Errors, error) {
...
+type MoveResult struct {
+	valid     bool
+	errorCode FetchError
+	move      Move
+}
+
+// IsValid indicates wether the MoveResult is valid.
```
Comment (amitch23) [inline pkg/models/move.go]: "whether"

File: pkg/models/move.go [hunk 3]
```diff
@@ -56,11 +60,73 @@ func (m *Move) ValidateUpdate(tx *pop.Connection) (*validate.Errors, error) {
...
+// Move returns the move if and only if the move was correctly fetched
+func (m MoveResult) Move() Move {
+	if !m.valid {
+		fmt.Println("Check if this isValid before accessing the Move()!")
+		os.Exit(1)
+	}
+	return m.move
+}
+
+// ErrorCode returns the error if and only iff the move was not correctly fetched
```
Comment (amitch23) [inline pkg/models/move.go]: sp. "if"

File: swagger/internal.yaml [hunk 4]
```diff
@@ -351,24 +359,29 @@ definitions:
...
+        x-nullable: true
       created_at:
         type: string
         format: date-time
+        x-nullable: true
       updated_at:
         type: string
         format: date-time
+        x-nullable: true
   PatchPersonallyProcuredMovePayload:
```
Comment (amitch23) [inline swagger/internal.yaml]: Why have both patch and update?

File: pkg/models/move.go [hunk 5]
```diff
@@ -56,11 +60,73 @@ func (m *Move) ValidateUpdate(tx *pop.Connection) (*validate.Errors, error) {
...
+
+// IsValid indicates wether the MoveResult is valid.
+func (m MoveResult) IsValid() bool {
+	return m.valid
+}
+
+// Move returns the move if and only if the move was correctly fetched
+func (m MoveResult) Move() Move {
+	if !m.valid {
+		fmt.Println("Check if this isValid before accessing the Move()!")
```
Comment (amitch23) [inline pkg/models/move.go]: should we be logging or printing this? (same question for other print lines...)

File: swagger/internal.yaml (hunk 4 above)
Comment (macrael) [reply swagger/internal.yaml]: It's specified in ADR#13, update allows you to update an entire record at once, and patch allows you to selectively update specific fields

File: pkg/models/move.go (hunk 5 above)
Comment (macrael) [reply pkg/models/move.go]: Logging, I'll fix it.

File: pkg/models/errors.go [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+
+// FetchError is a base type that can typecast for specific APIs,
+// It indicicates why an attempted db fetch failed.
+type FetchError string
+
+const (
+	// FetchErrorNotFound means that the requested record does not exist
+	FetchErrorNotFound FetchError = "NOT_FOUND"
+	// FetchErrorForbidden means that the record exists but that the user does not have access to it
+	FetchErrorForbidden FetchError = "NOT_AUTHORIZED"
```
Comment (stangah) [inline pkg/models/errors.go]: forbidden

**Pull request merged**